rfcs/0004-leader-rotation.md

```diff
@@ -1,104 +1,29 @@
 # Leader Rotation
 
-The goal of this RFC is to define how leader nodes are rotated in Solana, how rotation may cause forks to arise, and how the converges
-in response.
+The goal of this RFC is to define how leader nodes are rotated in Solana.
 
-## Leader Seed Generation
+## Leader Schedule Generation
 
-Leader selection is decided via a random seed.  The process is as follows:
+Leader schedule is decided via a predefined seed.  The process is as follows:
 
-1. Periodically at a specific `PoH tick count` select the first vote signatures that create a supermajority from the previous voting round.
-2. Append them together.
-3. Hash the string for `N` counts via a similar process as PoH itself.
-4. The resulting hash is the random seed for `M` counts, `M` leader periods, where M > N
+1. Periodically at a specific `PoH tick count` use the tick count (simple monotonically increasing counter) as a seed to a stable psudo-random algorithm.
+2. At that height, compute all the currently staked accounts and their assigned leader identities and weights.
+3. Sort them by stake weight.
+4. Using the random seed select nodes weighted by stake to create a stake weighted ordering.
+5. This ordering becomes valid in `N` `PoH tick counts`.
```

Comment:
That doesn't seem right.  Seems like it ought to be something like N counts after the start of the slot.

Reply:
the leader schedule needs to be available for a good while before it comes into effect to reduce forking that might arise because schedules themselves have forked

Reply:
ideally, I'd want to see leader schedules available no later than 32 slots before they come into effect...